### PR TITLE
drivers: serial: nrf: Use flow control from device tree

### PR DESCRIPTION
**Pull request by nordic-krch**
Removed flow control configuration from Kconfig and started to use device tree. Added option to have single direction flow control with only rts or cts pin.

### CODE REVIEW

File: drivers/serial/uart_nrfx_uarte.c [hunk 1]
```diff
@@ -1452,8 +1464,8 @@ static int uarte_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
 		    (UARTE_PROP(idx, pin_prop)),			       \
 		    (NRF_UARTE_PSEL_DISCONNECTED))
 
-#define UARTE_RTS_CTS_PINS_SET(idx)					       \
-	(UARTE_HAS_PROP(idx, rts_pin) && UARTE_HAS_PROP(idx, cts_pin))
+#define UARTE_RTS_OR_CTS_PINS_SET(idx)					       \
```
Comment (anangl) [inline drivers/serial/uart_nrfx_uarte.c:1467]: What about naming it `FLOW_CONTROL_AVAILABLE`?

File: drivers/serial/uart_nrfx_uarte.c [hunk 2]
```diff
@@ -1462,7 +1474,16 @@ static int uarte_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
 		irq_enable(DT_IRQN(UARTE(idx)));			       \
 	} while (0)
 
+#define HWFC_CONFIG_CHECK(idx) \
+	BUILD_ASSERT( \
+		(UARTE_PROP(idx, hw_flow_control) && \
+		(UARTE_HAS_PROP(idx, rts_pin) || UARTE_HAS_PROP(idx, cts_pin)))\
```
Comment (anangl) [inline drivers/serial/uart_nrfx_uarte.c:1480]: You could use `UARTE_RTS_OR_CTS_PINS_SET(idx)` here, for better readability. This line should be also indented more.

File: boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts [hunk 3]
```diff
@@ -140,7 +140,7 @@
 };
 
 &uart0 {
-	compatible = "nordic,nrf-uart";
+	compatible = "nordic,nrf-uarte";
```
Comment (anangl) [inline boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts:143]: Was that intentional? If so, it should be mentioned in the commit message, or perhaps better done in a separate commit.

**Review by joerchan — CHANGES_REQUESTED** [review]
HCI uart sample has not enabled HW FC for bbc_microbit board.

**Comment by nordic-krch** [comment]
> HCI uart sample has not enabled HW FC for bbc_microbit board.

what do you mean? DTS for bbc-micro has HWFC off for uart0 and sample does not modify it so it will have hwfc off.

File: boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts (hunk 3 above)
Comment (nordic-krch) [reply boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts]: removed, it was accidentally added.

**Comment by nordic-krch** [comment]
@anangl can you re-review?

**Review by anangl — APPROVED** [review]

**Comment by joerchan** [comment]
> > HCI uart sample has not enabled HW FC for bbc_microbit board.
> 
> what do you mean? DTS for bbc-micro has HWFC off for uart0 and sample does not modify it so it will have hwfc off.

Right. HWFC is required for hci-uart though. I'm thinking we should actually remove the BBC microbit board configuration then.
Can take that outside of this PR though.

**Review by joerchan — APPROVED** [review]

**Review by Mierunski — APPROVED** [review]

**Comment by ioannisg** [comment]
@nordic-krch pls, rebase for conflict resolution
